src/AWS.Deploy.CLI/ServerMode/Controllers/RecipeController.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.\r
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Linq;
+using AWS.Deploy.CLI.ServerMode.Models;
+using AWS.Deploy.Orchestration;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace AWS.Deploy.CLI.ServerMode.Controllers
+{
+    [Produces("application/json")]
+    [ApiController]
+    [Route("api/v1/[controller]")]
+    public class RecipeController : ControllerBase
+    {
+        /// <summary>
+        /// Gets a summary of the specified Recipe.
+        /// </summary>
+        [HttpGet("{recipeId}")]
+        [SwaggerOperation(OperationId = "GetRecipe")]
+        [SwaggerResponse(200, type: typeof(RecipeSummary))]
+        public IActionResult GetRecipe(string recipeId)
+        {
+            if (string.IsNullOrEmpty(recipeId))
+            {
+                return NotFound($"A Recipe ID was not provided.");
```

Comment:
Instead of `NotFound` I think `BadRequest` makes more sense.

Reply:
Fixed